doc/source/workflow.rst

```diff
@@ -0,0 +1,726 @@
+       <package name="PACKAGE" bootinclude="true"/>
+     </packages>
+
+**How to add missing tools or libraries**
+  Additional software can be added to the boot image with the use of the
+  ``bootinclude`` attribute of the ``package`` or the ``archive`` element. At
+  the end of the boot image creation process kiwi attempts to reduce the size
+  of the boot image by removing files that are not part of a known list of
+  required files or their dependencies.
+
+  The list of required files is hard coded in the
+  :file:`<install-dir>kiwi/config/strip.xml` file. If you added files to the
+  boot image that are needed for your specific use case, you need to instruct
+  kiwi to not strip them from the image. This is accomplished by adding a
+  ``strip`` section to the system image :file:`config.xml` file, with the type
+  attribute set to tools, as follows:
+
+  .. code-block: xml
+
+     <strip type="tools"/>
+       <file name="FILENAME"/>
+     </strip>
+
+  The removal/preservation of files is name-based only, so you do not need to
+  specify a complete path, but rather the file name.
+
+**Is is possible to add boot code?**
+  Yes, as described in the `Boot Image Hook-Scripts`_ section above, KIWI
+  supports the execution of boot code at various times for various image types
+  using *hook* scripts. 
+
+**Is is possible to include completely customized boot code?**
+  No. In cases where the provided hooks are insufficient and the KIWI provided
+  boot code needs to be replaced completely, it is necessary to create a custom
+  boot image description. In this case, all parts of the boot image description
+  must be created by the user. It is best to use one of the KIWI provided boot
+  descriptions as a template. 
+
+**My customized boot image refuses to boot. How to debug?**
+  An initrd created by KIWI that is based on one of the KIWI- provided boot
+  image descriptions recognizes kernel parameters that are useful for debugging
+  purposes, in case the image does not boot. These parameters may not work if
+  the image contains a custom boot image where the kiwi boot code has been
+  completely replaced. Some hints are described in `Boot Debugging`_ section.
+
+
+Distribution Specific Code
+--------------------------
+
+KIWI is designed to be distribution-independent. However, Linux distributions
+differ from each other, primarily in the package management area and in the area
+of creation and composition of the boot image. Within the KIWI code base major
+areas of Linux distribution differences are isolated into specific regions of
+the code. The remainder of the code is common and distribution- independent. 
+
+KIWI-provided functions that are distribution-specific contain the distribution
+name as a prefix, such as ``suseStripKernel``. Scripts that are part of the boot
+code and are distribution-specific are identified by a prefix of the
+distribution name followed by a "-", for example ``suse-linuxrc``. When KIWI
```

Comment:
This does not apply to the v8 version of kiwi. The same kiwi boot code is used for all supported distributions. If it should be required to distinguish by the distro it only happens in the code. So there is a linuxrc but no suse-linuxrc. Same applies to repart vs suse-repart and all other kiwi initrd boot files. I suggest to delete that part from the docs here